src/pipes/db-lookup.pipe.ts

```diff
@@ -4,42 +4,31 @@ import {
   ValidationArguments,
 } from "class-validator";
 import { FindOneOptions } from "typeorm";
-import { DbLookupValidator } from '../validators';
+import { DbLookupValidator } from "../validators/db-lookup.validator";
+
+export interface DbLookupConfig {
+  type: any;
+  findOption?: ((validationArguments: ValidationArguments) => FindOneOptions) | "primary";
+  validateNumeric?: boolean;
+  ignoreEmpty?: boolean;
+}
 
 export function DbLookup(
-  type: any,
-  findOption?:
-    | ((validationArguments: ValidationArguments) => FindOneOptions)
-    | "primary"
-    | {
-        validateNumeric?: boolean;
-        ignoreEmpty?: boolean;
-        findOption?: ((validationArguments: ValidationArguments) => FindOneOptions) | "primary";
-      },
-  validationOptions?: ValidationOptions,
-  ignoreEmpty = true
+  config: DbLookupConfig,
+  validationOptions?: ValidationOptions
```

Comment:
Related to the comment above, this breaks the current interface for the pipe. The `ignoreEmpty` parameter, however, is not used in any place the pipe is used, so I think we could replace that parameter:
```suggestion
  type: any,
  findOption?: ((validationArguments: ValidationArguments) => FindOneOptions) | "primary",
  validationOptions?: ValidationOptions,
  config: DbLookupConfig,
```

Reply:
Pls see PR #170.